### ISSUE TITLE
JeanLucPons a dev which listens to the community

### ISSUE THREAD

**Issue by SatoshiNakamotoBitcoins** [JeanLucPons/VanitySearch#10]
Great app...frequent updates...pays attention to idea's of the community!!

JeanLucPons the new kid on the block...;-)

**Comment by JeanLucPons** [comment]
:+1: 
:)

**Comment by dem10** [comment]
It is a pity that the VanitySearch does not work with AMD cards.Other than that, great program.

**Comment by JeanLucPons** [comment]
I'm not familiar with OpenCL and I don't know how to program efficient bigint calculations without having intrinsic function to access to the carry flag (for add and sub), this is the main annoying point.  I can use of trick to workaround this but it is not very efficient and performance depend a lot on the OpenCL compiler.

**Comment by dem10** [comment]
> I'm not familiar with OpenCL and I don't know how to program efficient bigint calculations without having intrinsic function to access to the carry flag (for add and sub), this is the main annoying point. I can use of trick to workaround this but it is not very efficient and performance depend a lot on the OpenCL compiler.

I'm not a programmer,but it would be great if people with AMD cards could connect not only the processor.Perhaps the original oclvanitygen will give you some answers.But it has very little speed on new drivers.

**Comment by JeanLucPons** [comment]
> I'm not a programmer,but it would be great if people with AMD cards could connect not only the processor.Perhaps the original oclvanitygen will give you some answers.But it has very little speed on new drivers.
I agree. I'm trying to find efficient solution to implement bigint calculations with OpenCL but I don't find much thing. I know more or less the OpenCL code of oclvanitygen. It uses tricks to handle carry but as I said it is not very efficient and it depends on how the driver will compile this code. Some drivers may generate the good code using the carry flag, others will generate ugly and terribly slow conditional branches. Integer arithmetic is quite a crucial part in cryptography.

**Comment by SatoshiNakamotoBitcoins** [comment]
@dem10 Grep some BTC with Hamnaz's list and the new repo's of The Real Lord and get some RTX 2080 Ti's...;-)

Cutting the chase...maybe there will be a smart coder which will fork Vanitysearch and applies OpenCL...give time some time...

Cheers

**Comment by JeanLucPons** [comment]
Here is the code of oclvanitygen to perform an addition with carry:

```
#define bn_addc_word(r, a, b, t, c) do {			\
		t = a + b + c;					\
		c = (t < a) ? 1 : ((c & (t == a)) ? 1 : 0);	\
		r = t;						\
} while (0)
```

This can be reduced to a single instruction with CUDA (and also with Visual C++, gcc, etc...) !
Some OpenCL driver compilers are smart enough to understand this code are reduce it to a single adc instruction but not all !

**Comment by dem10** [comment]
> @dem10 Grep some BTC with Hamnaz's list and the new repo's of The Real Lord and get some RTX 2080 Ti's...;-)
> 
So far gtx 1080ti shows the best result.Perhaps the potential of rtx 2080ti is not disclosed in full.I have no possibility to buy the rtx 2080ti(but if you donate me some bitcoins,I will be able  to buy 2080ti or 1080ti 😁) while I'm content with what I have r9 390x.😪

**Comment by SatoshiNakamotoBitcoins** [comment]
@dem10 Quality over Quantity...(or a combination of that...;-))
